File: app/src/main/java/me/miran/mhdstuff/activity/MainActivity.java
package me.miran.mhdstuff.activity;

import android.os.Bundle;
import android.widget.Button;

import me.miran.mhdstuff.R;
import me.miran.mhdstuff.activity.base.BaseActivity;
import me.miran.mhdstuff.activity.settings.SettingsActivity;
import me.miran.mhdstuff.parsing.storage.IdStorage;

public class MainActivity extends BaseActivity {


    public MainActivity() {
        super("Dpmb stuff", R.layout.activity_main);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        IdStorage.onActivity(this);

        Button buttonOpenSearch = findViewById(R.id.open_departures);
        Button buttonOpenConnections = findViewById(R.id.open_connections);
        Button buttonOpenVehicleMap = findViewById(R.id.open_map);
        Button buttonOpenEvents = findViewById(R.id.open_events);
        Button buttonOpenVehiclesList = findViewById(R.id.open_vehicles);
//        Button buttonOpenNews = findViewById(R.id.open_news);
        Button buttonOpenDiversions = findViewById(R.id.open_diversions);
        Button buttonOpenSettings = findViewById(R.id.open_settings);
        Button buttonOpenAbout = findViewById(R.id.open_about);

        buttonOpenSearch.setOnClickListener(v -> startActivity(SearchActivity.class));
//        buttonOpenNews.setOnClickListener(v -> startActivity(NewsActivity.class));
        buttonOpenConnections.setOnClickListener(v -> startActivity(ConnectionsResultActivity.class));
        buttonOpenVehicleMap.setOnClickListener(v -> startActivity(VehicleMapActivity.class));
        buttonOpenVehiclesList.setOnClickListener(v -> startActivity(VehiclesListActivity.class));
        buttonOpenSettings.setOnClickListener(v -> startActivity(SettingsActivity.class));
        buttonOpenDiversions.setOnClickListener(view -> startActivity(DiversionsActivity.class));
        buttonOpenEvents.setOnClickListener(view -> startActivity(EventsActivity.class));
        buttonOpenAbout.setOnClickListener(view -> startActivity(AboutActivity.class));
    }
}

File: app/src/main/java/me/miran/mhdstuff/activity/MainActivity.kt
package me.miran.mhdstuff.activity

import androidx.compose.runtime.Composable
import me.miran.mhdstuff.R
import me.miran.mhdstuff.activity.base.NavigationActivity
import me.miran.mhdstuff.activity.settings.SettingsActivity

class MainActivity : NavigationActivity(R.string.app_name) {


    @Composable
    override fun CreateNavigation() {
        NavigationItem(R.drawable.bus_light_full, R.string.departures, SearchActivity::class)
        NavigationItem(R.drawable.location_arrow, R.string.vehicle_map, VehicleMapActivity::class, true)
        NavigationItem(R.drawable.list_ul_regular, R.string.vehicles, VehiclesListActivity::class, true)
        NavigationItem(R.drawable.bolt_regular, R.string.events, EventsActivity::class, true)
        NavigationItem(R.drawable.triangle_exclamation_regular, R.string.diversions, DiversionsActivity::class, true)
        NavigationItem(R.drawable.address_card_regular, "Šalinkarta") {}
        NavigationItem(R.drawable.code_fork_regular, R.string.schemes) {}
        NavigationItem(R.drawable.gear_regular, R.string.settings, SettingsActivity::class)
        NavigationItem(R.drawable.circle_info_regular, R.string.about, AboutActivity::class)
    }



}